server/src/metadata/data-source/data-source.service.ts

```diff
@@ -121,6 +121,20 @@ export class DataSourceService implements OnModuleInit, OnModuleDestroy {
     return `workspace_${uuidToBase36(workspaceId)}`;
   }
 
+  public async deleteWorkspaceSchema(workspaceId: string) {
+    const schemaName = this.getSchemaName(workspaceId);
+    const queryRunner = this.mainDataSource.createQueryRunner();
+    const schemaAlreadyExists = await queryRunner.hasSchema(schemaName);
+
+    if (!schemaAlreadyExists) {
+      throw new Error(`Schema ${schemaName} does not exist`);
```

Comment:
@martmull let's add a `await queryRunner.release()` here, I believe we have the same issue in the createSchema method